File: app/src/main/java/com/cleanup/todoc/datasource/Entity/ProjectWithTasks.java
package com.cleanup.todoc.datasource.Entity;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

/*
 * Representation Room of the relation one(Project) to many(Task)
 */
public class ProjectWithTasks {
    @Embedded
    public ProjectEntity projectEntity;
    @Relation(
            parentColumn = "id",
            entityColumn = "projectId"
    )
    public List<TaskEntity> taskEntities;
}

File: app/src/main/java/com/cleanup/todoc/datasource/entity/ProjectWithTasks.kt
package com.cleanup.todoc.datasource.entity

import androidx.room.Embedded
import androidx.room.Relation

/*
 * Representation Room of the relation one(Project) to many(Task)
 */
class ProjectWithTasks {
    @JvmField
    @Embedded
    var projectEntity: ProjectEntity? = null

    @JvmField
    @Relation(parentColumn = "id", entityColumn = "projectId")
    var taskEntities: List<TaskEntity>? = null
}
